src/core_write.cpp

```diff
@@ -88,7 +88,12 @@ string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDecode)
         }
         if (0 <= opcode && opcode <= OP_PUSHDATA4) {
             if (vch.size() <= static_cast<vector<unsigned char>::size_type>(4)) {
-                str += strprintf("%d", CScriptNum(vch, false).getint());
+                if ( (lastopcode == OP_NAME_NEW || lastopcode == OP_NAME_UPDATE || lastopcode == OP_NAME_FIRSTUPDATE )
+                    && vch.size() > static_cast<vector<unsigned char>::size_type>(0) ) {
```

Comment:
No particular reason except to maintain parallelism with the other check. I think it might be a failure of the class to show the vch as empty() when in reality there's a 0x00 in there.